### PR TITLE
Close gRPC client in tests

### PR DESCRIPTION
**Pull request by ignasi35**
References #577

Pending: 

- [x] replicate to scala code base, 
- [x] replicate to other tests in the same service and other services in the microservices tutorial and other modules (e.g. `how-to/`), 
- [x] copy/paste code to all versions of the sample code.

### CODE REVIEW

**Comment by ignasi35** [comment]
This continually fails on #578 and #591

**Review by patriknw — COMMENTED** [review]
good

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/main/java/shopping/cart/Main.java [hunk 1]
```diff
@@ -65,6 +66,14 @@ static ShoppingOrderService orderServiceClient(ActorSystem<?> system) {
...
             .withTls(false);
 
-    return ShoppingOrderServiceClient.create(orderServiceClientSettings, system);
+    ShoppingOrderServiceClient client =
+        ShoppingOrderServiceClient.create(orderServiceClientSettings, system);
+    CoordinatedShutdown.get(system)
+        .addTask(
+            CoordinatedShutdown.PhaseBeforeServiceUnbind(),
+            "close-test-client-for-grpc",
+            () -> client.close());
```
Comment (patriknw) [inline docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/main/java/shopping/cart/Main.java:75]: Is this needed? Does it solve anything over letting it be shutdown when the ActorSystem terminates? Is it "Pool shutdown unexpectedly" here also, not only in tests?

The reason I'm asking is that I don't want us to more infra structure code than necessary?

Could it be a possible improvement in Akka HTTP or gRCP to shutdown more clean on ordinary ActorSystem termination?

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/main/java/shopping/cart/Main.java [hunk 2]
```diff
@@ -65,6 +66,14 @@ static ShoppingOrderService orderServiceClient(ActorSystem<?> system) {
                 system)
             .withTls(false);
 
-    return ShoppingOrderServiceClient.create(orderServiceClientSettings, system);
+    ShoppingOrderServiceClient client =
+        ShoppingOrderServiceClient.create(orderServiceClientSettings, system);
+    CoordinatedShutdown.get(system)
+        .addTask(
+            CoordinatedShutdown.PhaseBeforeServiceUnbind(),
```
Comment (patriknw) [inline docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/main/java/shopping/cart/Main.java:73]: This phase might be too early. Client might sill be in used, projections haven't been stopped yet.

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/test/java/shopping/cart/IntegrationTest.java [hunk 3]
```diff
@@ -107,6 +108,11 @@ public TestNodeFixture(int grcpPort, List<Integer> managementPorts, int manageme
     public shopping.cart.proto.ShoppingCartService getClient() {
       if (client == null) {
         client = shopping.cart.proto.ShoppingCartServiceClient.create(clientSettings, system);
+        CoordinatedShutdown.get(system)
+            .addTask(
+                CoordinatedShutdown.PhaseBeforeServiceUnbind(),
```
Comment (patriknw) [inline docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/test/java/shopping/cart/IntegrationTest.java:113]: isn't it better to close it as late as possible?

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/main/java/shopping/cart/Main.java (hunk 1 above)
Comment (ignasi35) [reply docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/main/java/shopping/cart/Main.java:75]: I was on cruise control but I think this not right. This gRPC client is part of a projection that will be stopped when leaving the cluster. I'll reset this change.

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/main/java/shopping/cart/Main.java (hunk 2 above)
Comment (ignasi35) [reply docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/main/java/shopping/cart/Main.java:73]: Yeah. I've stepped back. I think that as long as this client is only used in the projection it's OK if we don't register a task to shutdown the client.

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/test/java/shopping/cart/IntegrationTest.java (hunk 3 above)
Comment (ignasi35) [reply docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/test/java/shopping/cart/IntegrationTest.java:113]: I'm not sure. My thinking was:

 - if the server unbinds before the client has shut down, any open connection will be dropped

Comment (patriknw) [reply docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/src/test/java/shopping/cart/IntegrationTest.java:113]: ok, this is the client that is used by test itself. good to close that as early as possible.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**